ceno_zkvm/src/bin/e2e.rs

```diff
@@ -94,6 +100,12 @@ fn main() {
     let elf_bytes = fs::read(&args.elf).expect("read elf file");
     let mut vm = VMState::new_from_elf(platform.clone(), &elf_bytes).unwrap();
 
+    tracing::info!("Loading private input file: {:?}", args.private_input);
+    let priv_io = memory_from_file(&args.private_input);
+    for (addr, value) in zip(platform.private_io.iter_addresses(), &priv_io) {
+        vm.init_memory(addr.into(), *value);
+    }
```

Comment:
should we add a comment to say, if len(private_io) != len(private_input_file), then it is right-padded with 0s?

Reply:
Good point. It is **not** padded. Or precisely it is zero-padded to the next power-of-two size.

The rest of memory does not exist - unsatisfiable if the program tries to use it.

Reply:
For technical reasons, [if we are using rkyv, it would be better to pad from the left by default.](https://github.com/scroll-tech/ceno/discussions/607)

That's means the last byte that's specified should always appear in the same position in the VM.

> The rest of memory does not exist - unsatisfiable if the program tries to use it.

That's good.